### PR TITLE
Cpu fusions using post_ops

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/targets/cpu/fuse_ops.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+MIGRAPHX_PRED_MATCHER(without_post_ops, instruction_ref ins)
+{
+    auto v = ins->get_operator().to_value();
+    return v.contains("post_ops") and v["post_ops"].empty();
+}
+
+bool workaround_dnnl_broken_post_ops(const operation& op, const operation& post_op)
+{
+    if(contains({"dnnl::dot", "dnnl::convolution"}, op.name()))
```
Comment (kahmed10) [inline src/targets/cpu/fuse_ops.cpp:31]: so all convolutions and gemm fusions are broken currently?

Comment (pfultz2) [reply src/targets/cpu/fuse_ops.cpp:31]: They most likely need to use the 3 argument form. When using an 'add' post_op for convolution or dot it does work correctly but it usually falls back on a ref implementation which is incredibly slow. Other post_ops don't seem to work consistently.

This restricts it to only use post_ops when it is binary+eltwise. It does work in some other scenarios but its hard to characterize when it works and when it doesn't.

Comment (kahmed10) [reply src/targets/cpu/fuse_ops.cpp:31]: I am not sure that we should call the function `workaround_dnnl_broken_post_ops` if it's just op+eltwise. Maybe something like `check_op_eltwise`? Within the function, we could also include a comment about why these fusions aren't supported currently.

Comment (pfultz2) [reply src/targets/cpu/fuse_ops.cpp:31]: It is a workaround. dnnl shouldn't return an algorithm when the set of ops are not supported. I would prefer to keep the name whereas `check_op_eltwise` name would not be accurate if we refine this workaround in the future.

**Review by kahmed10 — APPROVED** [review]

**Review by scxiao — APPROVED** [review]

File: src/include/migraphx/op/broadcast.hpp [hunk 2]
```diff
@@ -66,6 +66,7 @@ struct broadcast
     {
         return {std::move(output_shape), std::move(args.at(0).data)};
     }
+    bool is_borrowed() const { return true; }
```
[the remark this answers is not in the document]
Comment (pfultz2) [reply src/include/migraphx/op/broadcast.hpp:69]: It means it uses a weak reference to the memory in the inputs(ie constructing an argument from `.data()` of another argument).

**Pull request merged**